src/Nix/Options.hs

```diff
@@ -35,6 +37,7 @@ data Options = Options
     , arg          :: [(Text, Text)]
     , argstr       :: [(Text, Text)]
     , fromFile     :: Maybe FilePath
+    , currentTime  :: Maybe UTCTime
```

Comment:
Ah, I see, it's a `Maybe` here so that the user doesn't have to specify it on the command-line.  That's the right type, we just have to make sure that if anyone ever uses the current time, it's an error if it was never set.

Reply:
Actually, never mind that, we have it be just `UTCTime`, and then have both `defaultOptions` and `nixOptions` take a time value to use as the default if it's not specified.